[glib] Build error

Package: glib[core]:x86-windows -> 2.75.1

**Host Environment**

- Host: x64-windows
- Compiler: MSVC 19.29.30147.0
-    vcpkg-tool version: 2022-12-14-7ae0d8527fb488fde10a89c2813802dc9b03b6f9
    vcpkg-scripts version: 0f719b3 2023-01-03 (21 hours ago)

**To Reproduce**

`vcpkg install glib`

**Failure logs**

```
-- Downloading https://download.gnome.org/sources/glib/2.75/glib-2.75.1.tar.xz -> glib-2.75.1.tar.xz...
-- Extracting source D:/a/fheroes2-prebuilt-deps/fheroes2-prebuilt-deps/.vcpkg/downloads/glib-2.75.1.tar.xz
-- Applying patch use-libiconv-on-windows.patch
-- Applying patch libintl.patch
-- Using source at D:/a/fheroes2-prebuilt-deps/fheroes2-prebuilt-deps/.vcpkg/buildtrees/glib/src/glib-2-cc73e48deb.clean
-- Found external ninja('1.10.2').
-- Getting CMake variables for x86-windows
-- Downloading https://www.python.org/ftp/python/3.10.7/python-3.10.7-embed-win32.zip -> python-3.10.7-embed-win32.zip...
-- Configuring x86-windows-dbg
-- Using cached msys-mingw-w64-i686-pkg-config-0.29.2-3-any.pkg.tar.zst.
-- Using cached msys-mingw-w64-i686-libwinpthread-git-9.0.0.6373.5be8fcd83-1-any.pkg.tar.zst.
-- Using msys root at D:/a/fheroes2-prebuilt-deps/fheroes2-prebuilt-deps/.vcpkg/downloads/tools/msys2/9a1ec3f33446b195
-- Configuring x86-windows-dbg done
-- Configuring x86-windows-rel
-- Configuring x86-windows-rel done
-- Package x86-windows-dbg
-- Package x86-windows-rel
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:96 (message):
    Command failed: "C:/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/Common7/IDE/CommonExtensions/Microsoft/CMake/Ninja/ninja.exe" install -v
    Working Directory: D:/a/fheroes2-prebuilt-deps/fheroes2-prebuilt-deps/.vcpkg/buildtrees/glib/x86-windows-rel
    Error code: 1
    See logs for more information:
      D:\a\fheroes2-prebuilt-deps\fheroes2-prebuilt-deps\.vcpkg\buildtrees\glib\package-x86-windows-rel-out.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_install_meson.cmake:33 (vcpkg_execute_required_process)
  ports/glib/portfile.cmake:44 (vcpkg_install_meson)
  scripts/ports.cmake:147 (include)



```
<details><summary>D:\a\fheroes2-prebuilt-deps\fheroes2-prebuilt-deps\.vcpkg\buildtrees\glib\package-x86-windows-rel-out.log</summary>

```
[1/479] "D:/a/fheroes2-prebuilt-deps/fheroes2-prebuilt-deps/.vcpkg/downloads/tools/python/python-3.10.7-x86/python.exe" "../src/glib-2-cc73e48deb.clean/gio/data-to-c.py" "../src/glib-2-cc73e48deb.clean/gio/../glib/gconstructor.h" "gconstructor_code" "gio/gconstructor_as_data.h"
[2/479] "C:/Program Files (x86)/Windows Kits/10/bin/10.0.22621.0/x64/rc.exe" "/nologo" "/foglib/glib_glib.rc_glib.res" "glib/glib.rc"
[3/479] "D:\a\fheroes2-prebuilt-deps\fheroes2-prebuilt-deps\.vcpkg\downloads\tools\python\python-3.10.7-x86\python.exe" "D:/a/fheroes2-prebuilt-deps/fheroes2-prebuilt-deps/.vcpkg/buildtrees/glib/src/glib-2-cc73e48deb.clean/tools/gen-visibility-macros.py" "2.75.1" "visibility-macros" "GLIB" "glib/glib-visibility.h"
[4/479] "C:/Program Files (x86)/Windows Kits/10/bin/10.0.22621.0/x64/rc.exe" "/nologo" "/fogobject/gobject_gobject.rc_gobject.res" "gobject/gobject.rc"
[5/479] "D:\a\fheroes2-prebuilt-deps\fheroes2-prebuilt-deps\.vcpkg\downloads\tools\python\python-3.10.7-x86\python.exe" "D:/a/fheroes2-prebuilt-deps/fheroes2-prebuilt-deps/.vcpkg/buildtrees/glib/src/glib-2-cc73e48deb.clean/tools/gen-visibility-macros.py" "2.75.1" "visibility-macros" "GOBJECT" "gobject/gobject-visibility.h"
[6/479] "C:/Program Files (x86)/Windows Kits/10/bin/10.0.22621.0/x64/rc.exe" "/nologo" "/fogmodule/gmodule_gmodule.rc_gmodule.res" "gmodule/gmodule.rc"
[7/479] "D:\a\fheroes2-prebuilt-deps\fheroes2-prebuilt-deps\.vcpkg\downloads\tools\python\python-3.10.7-x86\python.exe" "D:/a/fheroes2-prebuilt-deps/fheroes2-prebuilt-deps/.vcpkg/buildtrees/glib/src/glib-2-cc73e48deb.clean/tools/gen-visibility-macros.py" "2.75.1" "visibility-macros" "GMODULE" "gmodule/gmodule-visibility.h"
[8/479] "C:/Program Files (x86)/Windows Kits/10/bin/10.0.22621.0/x64/rc.exe" "/nologo" "/fogthread/gthread_gthread.rc_gthread.res" "gthread/gthread.rc"
[9/479] "C:/Program Files (x86)/Windows Kits/10/bin/10.0.22621.0/x64/rc.exe" "/nologo" "/fogio/gio_gio.rc_gio.res" "gio/gio.rc"
[10/479] "D:\a\fheroes2-prebuilt-deps\fheroes2-prebuilt-deps\.vcpkg\downloads\tools\python\python-3.10.7-x86\python.exe" "D:\a\fheroes2-prebuilt-deps\fheroes2-prebuilt-deps\.vcpkg\installed\x64-windows\tools\meson\meson.py" "--internal" "exe" "--capture" "gobject\glib-enumtypes.h" "--" "D:/a/fheroes2-prebuilt-deps/fheroes2-prebuilt-deps/.vcpkg/downloads/tools/python/python-3.10.7-x86/python.exe" "gobject/glib-mkenums" "--template" "../src/glib-2-cc73e48deb.clean/gobject/glib-enumtypes.h.template" "../src/glib-2-cc73e48deb.clean/gobject/../glib/gunicode.h"
[11/479] "D:\a\fheroes2-prebuilt-deps\fheroes2-prebuilt-deps\.vcpkg\downloads\tools\python\python-3.10.7-x86\python.exe" "D:\a\fheroes2-prebuilt-deps\fheroes2-prebuilt-deps\.vcpkg\installed\x64-windows\tools\meson\meson.py" "--internal" "exe" "--capture" "gio\gioenumtypes.h" "--" "D:/a/fheroes2-prebuilt-deps/fheroes2-prebuilt-deps/.vcpkg/downloads/tools/python/python-3.10.7-x86/python.exe" "gobject/glib-mkenums" "--template" "../src/glib-2-cc73e48deb.clean/gio/
...
Skipped 8 lines
...

"C:/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/VC/Tools/MSVC/14.29.30133/bin/Hostx64/x86/cl.exe" "-Igio\win32\libgiowin32.a.p" "-Igio\win32" "-I..\src\glib-2-cc73e48deb.clean\gio\win32" "-I." "-I..\src\glib-2-cc73e48deb.clean" "-Iglib" "-I..\src\glib-2-cc73e48deb.clean\glib" "-Igio" "-I..\src\glib-2-cc73e48deb.clean\gio" "-Igmodule" "-I..\src\glib-2-cc73e48deb.clean\gmodule" "-Igobject" "-ID:/a/fheroes2-prebuilt-deps/fheroes2-prebuilt-deps/.vcpkg/installed/x86-windows/include" "/MD" "/nologo" "/showIncludes" "/utf-8" "/W4" "/Od" "/FImsvc_recommended_pragmas.h" "/utf-8" "-D_GNU_SOURCE" "-DG_DISABLE_CAST_CHECKS" "/wo4057" "/wd4068" "/wo4090" "/wd4100" "/wd4116" "/wo4125" "/wd4127" "/wd4146" "/wd4152" "/wd4201" "/wd4232" "/wo4245" "/wo4267" "/wd4334" "/wo4389" "/wo4702" "/wd4706" "-nologo" "-DWIN32" "-D_WINDOWS" "-W3" "-utf-8" "-MP" "-MD" "-O2" "-Oi" "-Gy" "-DNDEBUG" "-Z7" "-DG_LOG_DOMAIN=\"GLib-GIO\"" "-DGIO_LAUNCH_DESKTOP=\"D:/a/fheroes2-prebuilt-deps/fheroes2-prebuilt-deps/.vcpkg/packages/glib_x86-windows/libexec/gio-launch-desktop\"" "-DGIO_MODULE_DIR=\"D:/a/fheroes2-prebuilt-deps/fheroes2-prebuilt-deps/.vcpkg/packages/glib_x86-windows/lib/gio/modules\"" "-DLOCALSTATEDIR=\"D:/a/fheroes2-prebuilt-deps/fheroes2-prebuilt-deps/.vcpkg/packages/glib_x86-windows/var\"" "-DGIO_COMPILATION" "/Fdgio\win32\libgiowin32.a.p\gwin32filemonitor.c.pdb" /Fogio/win32/libgiowin32.a.p/gwin32filemonitor.c.obj "/c" ../src/glib-2-cc73e48deb.clean/gio/win32/gwin32filemonitor.c
cl : Command line warning D9025 : overriding '/W4' with '/W3'
cl : Command line warning D9025 : overriding '/Od' with '/O2'
D:\a\fheroes2-prebuilt-deps\fheroes2-prebuilt-deps\.vcpkg\buildtrees\glib\src\glib-2-cc73e48deb.clean\glib/gtypes.h(36): fatal error C1083: Cannot open include file: 'glib/gversionmacros.h': No such file or directory
[17/479] "C:/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/VC/Tools/MSVC/14.29.30133/bin/Hostx64/x86/cl.exe" "-Igio\win32\libgiowin32.a.p" "-Igio\win32" "-I..\src\glib-2-cc73e48deb.clean\gio\win32" "-I." "-I..\src\glib-2-cc73e48deb.clean" "-Iglib" "-I..\src\glib-2-cc73e48deb.clean\glib" "-Igio" "-I..\src\glib-2-cc73e48deb.clean\gio" "-Igmodule" "-I..\src\glib-2-cc73e48deb.clean\gmodule" "-Igobject" "-ID:/a/fheroes2-prebuilt-deps/fheroes2-prebuilt-deps/.vcpkg/installed/x86-windows/include" "/MD" "/nologo" "/showIncludes" "/utf-8" "/W4" "/Od" "/FImsvc_recommended_pragmas.h" "/utf-8" "-D_GNU_SOURCE" "-DG_DISABLE_CAST_CHECKS" "/wo4057" "/wd4068" "/wo4090" "/wd4100" "/wd4116" "/wo4125" "/wd4127" "/wd4146" "/wd4152" "/wd4201" "/wd4232" "/wo4245" "/wo4267" "/wd4334" "/wo4389" "/wo4702" "/wd4706" "-nologo" "-DWIN32" "-D_WINDOWS" "-W3" "-utf-8" "-MP" "-MD" "-O2" "-Oi" "-Gy" "-DNDEBUG" "-Z7" "-DG_LOG_DOMAIN=\"GLib-GIO\"" "-DGIO_LAUNCH_DESKTOP=\"D:/a/fheroes2-prebuilt-deps/fheroes2-prebuilt-deps/.vcpkg/packages/glib_x86-windows/libexec/gio-launch-desktop\"" "-DGIO_MODULE_DIR=\"D:/a/fheroes2-prebuilt-deps/fheroes2-prebuilt-deps/.vcpkg/packages/glib_x86-windows/lib/gio/modules\"" "-DLOCALSTATEDIR=\"D:/a/fheroes2-prebuilt-deps/fheroes2-prebuilt-deps/.vcpkg/packages/glib_x86-windows/var\"" "-DGIO_COMPILATION" "/Fdgio\win32\libgiowin32.a.p\gwinhttpfile.c.pdb" /Fogio/win32/libgiowin32.a.p/gwinhttpfile.c.obj "/c" ../src/glib-2-cc73e48deb.clean/gio/win32/gwinhttpfile.c
FAILED: gio/win32/libgiowin32.a.p/gwinhttpfile.c.obj 
"C:/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/VC/Tools/MSVC/14.29.30133/bin/Hostx64/x86/cl.exe" "-Igio\win32\libgiowin32.a.p" "-Igio\win32" "-I..\src\glib-2-cc73e48deb.clean\gio\win32" "-I." "-I..\src\glib-2-cc73e48deb.clean" "-Iglib" "-I..\src\glib-2-cc73e48deb.clean\glib" "-Igio" "-I..\src\glib-2-cc73e48deb.clean\gio" "-Igmodule" "-I..\src\glib-2-cc73e48deb.clean\gmodule" "-Igobject" "-ID:/a/fheroes2-prebuilt-deps/fheroes2-prebuilt-deps/.vcpkg/installed/x86-windows/include" "/MD" "/nologo" "/showIncludes" "/utf-8" "/W4" "/Od" "/FImsvc_recommended_pragmas.h" "/utf-8" "-D_GNU_SOURCE" "-DG_DISABLE_CAST_CHECKS" "/wo4057" "/wd4068" "/wo4090" "/wd4100" "/wd4116" "/wo4125" "/wd4127" "/wd4146" "/wd4152" "/wd4201" "/wd4232" "/wo4245" "/wo4267" "/wd4334" "/wo4389" "/wo4702" "/wd4706" "-nologo" "-DWIN32" "-D_WINDOWS" "-W3" "-utf-8" "-MP" "-MD" "-O2" "-Oi" "-Gy" "-DNDEBUG" "-Z7" "-DG_LOG_DOMAIN=\"GLib-GIO\"" "-DGIO_LAUNCH_DESKTOP=\"D:/a/fheroes2-prebuilt-deps/fheroes2-prebuilt-deps/.vcpkg/packages/glib_x86-windows/libexec/gio-launch-desktop\"" "-DGIO_MODULE_DIR=\"D:/a/fheroes2-prebuilt-deps/fheroes2-prebuilt-deps/.vcpkg/packages/glib_x86-windows/lib/gio/modules\"" "-DLOCALSTATEDIR=\"D:/a/fheroes2-prebuilt-deps/fheroes2-prebuilt-deps/.vcpkg/packages/glib_x86-windows/var\"" "-DGIO_COMPILATION" "/Fdgio\win32\libgiowin32.a.p\gwinhttpfile.c.pdb" /Fogio/win32/libgiowin32.a.p/gwinhttpfile.c.obj "/c" ../src/glib-2-cc73e48deb.clean/gio/win32/gwinhttpfile.c
cl : Command line warning D9025 : overriding '/W4' with '/W3'
cl : Command line warning D9025 : overriding '/Od' with '/O2'
D:\a\fheroes2-prebuilt-deps\fheroes2-prebuilt-deps\.vcpkg\buildtrees\glib\src\glib-2-cc73e48deb.clean\glib/gtypes.h(36): fatal error C1083: Cannot open include file: 'glib/gversionmacros.h': No such file or directory
[18/479] "D:\a\fheroes2-prebuilt-deps\fheroes2-prebuilt-deps\.vcpkg\downloads\tools\python\python-3.10.7-x86\python.exe" "D:\a\fheroes2-prebuilt-deps\fheroes2-prebuilt-deps\.vcpkg\installed\x64-windows\tools\meson\meson.py" "--internal" "exe" "--capture" "gio\gioenumtypes.c" "--" "D:/a/fheroes2-prebuilt-deps/fheroes2-prebuilt-deps/.vcpkg/downloads/tools/python/python-3.10.7-x86/python.exe" "gobject/glib-mkenums" "--template" "../src/glib-2-cc73e48deb.clean/gio/gioenumtypes.c.template" "../src/glib-2-cc73e48deb.clean/gio/gappinfo.h" "../src/glib-2-cc73e48deb.clean/gio/gasyncinitable.h" "../src/glib-2-cc73e48deb.clean/gio/gasyncresult.h" "../src/glib-2-cc73e48deb.clean/gio/gbufferedinputstream.h" "../src/glib-2-cc73e48deb.clean/gio/gbufferedoutputstream.h" "../src/glib-2-cc73e48deb.clean/gio/gbytesicon.h" "../src/glib-2-cc73e48deb.clean/gio/gcancellable.h" "../src/glib-2-cc73e48deb.clean/gio/gcontenttype.h" "../src/glib-2-cc73e48deb.clean/gio/gcharsetconverter.h" "../src/glib-2-cc73e48deb.clean/gio/gconverter.h" "../src/glib-2-cc73e48deb.clean/gio/gconverterinputstream.h" "../src/glib-2-cc73e48deb.clean/gio/gconverteroutputstream.h" "../src/glib-2-cc73e48deb.clean/gio/gdatagrambased.h" "../src/glib-2-cc73e48deb.clean/gio/gdatainputstream.h" "../src/glib-2-cc73e48deb.clean/gio/gdataoutputstream.h" "../src/glib-2-cc73e48deb.clean/gio/gdebugcontroller.h" "../src/glib-2-cc73e48deb.clean/gio/gdebugcontrollerdbus.h" "../src/glib-2-cc73e48deb.clean/gio/gdrive.h" "../src/glib-2-cc73e48deb.clean/gio/gemblem.h" "../src/glib-2-cc73e48deb.clean/gio/gemblemedicon.h" "../src/glib-2-cc73e48deb.clean/gio/gfile.h" "../src/glib-2-cc73e48deb.clean/gio/gfileattribute.h" "../src/glib-2-cc73e48deb.clean/gio/gfileenumerator.h" "../src/glib-2-cc73e48deb.clean/gio/gfileicon.h" "../src/glib-2-cc73e48deb.clean/gio/gfileinfo.h" "../src/glib-2-cc73e48deb.clean/gio/gfileinputstream.h" "../src/glib-2-cc73e48deb.clean/gio/gfilemonitor.h" "../src/glib-2-cc73e48deb.clean/gio/gfilenamecompleter.h" "../src/glib-2-cc73e48deb.clean/gio/gfileoutputstream.h" "../src/glib-2-cc73e48deb.clean/gio/gfileiostream.h" "../src/glib-2-cc73e48deb.clean/gio/gfilterinputstream.h" "../src/glib-2-cc73e48deb.clean/gio/gfilteroutputstream.h" "../src/glib-2-cc73e48deb.clean/gio/gicon.h" "../src/glib-2-cc73e48deb.clean/gio/ginetaddress.h" "../src/glib-2-cc73e48deb.clean/gio/ginetaddressmask.h" "../src/glib-2-cc73e48deb.clean/gio/ginetsocketaddress.h" "../src/glib-2-cc73e48deb.clean/gio/ginitable.h" "../src/glib-2-cc73e48deb.clean/gio/ginputstream.h" "../src/glib-2-cc73e48deb.clean/gio/gio.h" "../src/glib-2-cc73e48deb.clean/gio/gio-autocleanups.h" "../src/glib-2-cc73e48deb.clean/gio/gioenums.h" "../src/glib-2-cc73e48deb.clean/gio/gioerror.h" "../src/glib-2-cc73e48deb.clean/gio/giomodule.h" "../src/glib-2-cc73e48deb.clean/gio/gioscheduler.h" "../src/glib-2-cc73e48deb.clean/gio/giostream.h" "../src/glib-2-cc73e48deb.clean/gio/giotypes.h" "../src/glib-2-cc73e48deb.clean/gio/gloadableicon.h" "../src/glib-2-cc73e48deb.clean/gio/gmount.h" "../src/glib-2-cc73e48deb.clean/gio/gmemoryinputstream.h" "../src/glib-2-cc73e48deb.clean/gio/gmemorymonitor.h" "../src/glib-2-cc73e48deb.clean/gio/gmemoryoutputstream.h" "../src/glib-2-cc73e48deb.clean/gio/gmountoperation.h" "../src/glib-2-cc73e48deb.clean/gio/gnativesocketaddress.h" "../src/glib-2-cc73e48deb.clean/gio/gnativevolumemonitor.h" "../src/glib-2-cc73e48deb.clean/gio/gnetworkaddress.h" "../src/glib-2-cc73e48deb.clean/gio/gnetworkmonitor.h" "../src/glib-2-cc73e48deb.clean/gio/gnetworkservice.h" "../src/glib-2-cc73e48deb.clean/gio/goutputstream.h" "../src/glib-2-cc73e48deb.clean/gio/gpermission.h" "../src/glib-2-cc73e48deb.clean/gio/gpollableinputstream.h" "../src/glib-2-cc73e48deb.clean/gio/gpollableoutputstream.h" "../src/glib-2-cc73e48deb.clean/gio/gpollableutils.h" "../src/glib-2-cc73e48deb.clean/gio/gpowerprofilemonitor.h" "../src/glib-2-cc73e48deb.clean/gio/gproxy.h" "../src/glib-2-cc73e48deb.clean/gio/gproxyaddress.h" "../src/glib-2-cc73e48deb.clean/gio/gproxyaddressenumerator.h" "../src/glib-2-cc73e48deb.clean/gio/gproxyresolver.h" "../src/glib-2-cc73e48deb.clean/gio/gresolver.h" "../src/glib-2-cc73e48deb.clean/gio/gresource.h" "../src/glib-2-cc73e48deb.clean/gio/gseekable.h" "../src/glib-2-cc73e48deb.clean/gio/gsimpleasyncresult.h" "../src/glib-2-cc73e48deb.clean/gio/gsimpleiostream.h" "../src/glib-2-cc73e48deb.clean/gio/gsimplepermission.h" "../src/glib-2-cc73e48deb.clean/gio/gsimpleproxyresolver.h" "../src/glib-2-cc73e48deb.clean/gio/gsocket.h" "../src/glib-2-cc73e48deb.clean/gio/gsocketaddress.h" "../src/glib-2-cc73e48deb.clean/gio/gsocketaddressenumerator.h" "../src/glib-2-cc73e48deb.clean/gio/gsocketclient.h" "../src/glib-2-cc73e48deb.clean/gio/gsocketconnectable.h" "../src/glib-2-cc73e48deb.clean/gio/gsocketconnection.h" "../src/glib-2-cc73e48deb.clean/gio/gsocketcontrolmessage.h" "../src/glib-2-cc73e48deb.clean/gio/gsocketlistener.h" "../src/glib-2-cc73e48deb.clean/gio/gsocketservice.h" "../src/glib-2-cc73e48deb.clean/gio/gsrvtarget.h" "../src/glib-2-cc73e48deb.clean/gio/gsubprocess.h" "../src/glib-2-cc73e48deb.clean/gio/gsubprocesslauncher.h" "../src/glib-2-cc73e48deb.clean/gio/gtask.h" "../src/glib-2-cc73e48deb.clean/gio/gtcpconnection.h" "../src/glib-2-cc73e48deb.clean/gio/gtcpwrapperconnection.h" "../src/glib-2-cc73e48deb.clean/gio/gthemedicon.h" "../src/glib-2-cc73e48deb.clean/gio/gthreadedsocketservice.h" "../src/glib-2-cc73e48deb.clean/gio/gtlsbackend.h" "../src/glib-2-cc73e48deb.clean/gio/gtlscertificate.h" "../src/glib-2-cc73e48deb.clean/gio/gtlsclientconnection.h" "../src/glib-2-cc73e48deb.clean/gio/gtlsconnection.h" "../src/glib-2-cc73e48deb.clean/gio/gtlsdatabase.h" "../src/glib-2-cc73e48deb.clean/gio/gtlsfiledatabase.h" "../src/glib-2-cc73e48deb.clean/gio/gtlsinteraction.h" "../src/glib-2-cc73e48deb.clean/gio/gtlspassword.h" "../src/glib-2-cc73e48deb.clean/gio/gtlsserverconnection.h" "../src/glib-2-cc73e48deb.clean/gio/gdtlsconnection.h" "../src/glib-2-cc73e48deb.clean/gio/gdtlsclientconnection.h" "../src/glib-2-cc73e48deb.clean/gio/gdtlsserverconnection.h" "../src/glib-2-cc73e48deb.clean/gio/gunixconnection.h" "../src/glib-2-cc73e48deb.clean/gio/gunixcredentialsmessage.h" "../src/glib-2-cc73e48deb.clean/gio/gunixfdlist.h" "../src/glib-2-cc73e48deb.clean/gio/gunixsocketaddress.h" "../src/glib-2-cc73e48deb.clean/gio/gvfs.h" "../src/glib-2-cc73e48deb.clean/gio/gvolume.h" "../src/glib-2-cc73e48deb.clean/gio/gvolumemonitor.h" "../src/glib-2-cc73e48deb.clean/gio/gzlibcompressor.h" "../src/glib-2-cc73e48deb.clean/gio/gzlibdecompressor.h" "../src/glib-2-cc73e48deb.clean/gio/glistmodel.h" "../src/glib-2-cc73e48deb.clean/gio/gliststore.h" "../src/glib-2-cc73e48deb.clean/gio/gapplication.h" "../src/glib-2-cc73e48deb.clean/gio/gapplicationcommandline.h" "../src/glib-2-cc73e48deb.clean/gio/gactiongroup.h" "../src/glib-2-cc73e48deb.clean/gio/gactionmap.h" "../src/glib-2-cc73e48deb.clean/gio/gsimpleactiongroup.h" "../src/glib-2-cc73e48deb.clean/gio/gremoteactiongroup.h" "../src/glib-2-cc73e48deb.clean/gio/gactiongroupexporter.h" "../src/glib-2-cc73e48deb.clean/gio/gdbusactiongroup.h" "../src/glib-2-cc73e48deb.clean/gio/gaction.h" "../src/glib-2-cc73e48deb.clean/gio/gpropertyaction.h" "../src/glib-2-cc73e48deb.clean/gio/gsimpleaction.h" "../src/glib-2-cc73e48deb.clean/gio/gmenumodel.h" "../src/glib-2-cc73e48deb.clean/gio/gmenu.h" "../src/glib-2-cc73e48deb.clean/gio/gmenuexporter.h" "../src/glib-2-cc73e48deb.clean/gio/gdbusmenumodel.h" "../src/glib-2-cc73e48deb.clean/gio/gnotification.h" "../src/glib-2-cc73e48deb.clean/gio/gsettingsbackend.h" "../src/glib-2-cc73e48deb.clean/gio/gsettingsschema.h" "../src/glib-2-cc73e48deb.clean/gio/gsettings.h" "../src/glib-2-cc73e48deb.clean/gio/gdbusauthobserver.h" "../src/glib-2-cc73e48deb.clean/gio/gcredentials.h" "../src/glib-2-cc73e48deb.clean/gio/gdbusutils.h" "../src/glib-2-cc73e48deb.clean/gio/gdbuserror.h" "../src/glib-2-cc73e48deb.clean/gio/gdbusaddress.h" "../src/glib-2-cc73e48deb.clean/gio/gdbusconnection.h" "../src/glib-2-cc73e48deb.clean/gio/gdbusmessage.h" "../src/glib-2-cc73e48deb.clean/gio/gdbusnameowning.h" "../src/glib-2-cc73e48deb.clean/gio/gdbusnamewatching.h" "../src/glib-2-cc73e48deb.clean/gio/gdbusproxy.h" "../src/glib-2-cc73e48deb.clean/gio/gdbusintrospection.h" "../src/glib-2-cc73e48deb.clean/gio/gdbusmethodinvocation.h" "../src/glib-2-cc73e48deb.clean/gio/gdbusserver.h" "../src/glib-2-cc73e48deb.clean/gio/gdbusinterface.h" "../src/glib-2-cc73e48deb.clean/gio/gdbusinterfaceskeleton.h" "../src/glib-2-cc73e48deb.clean/gio/gdbusobject.h" "../src/glib-2-cc73e48deb.clean/gio/gdbusobjectskeleton.h" "../src/glib-2-cc73e48deb.clean/gio/gdbusobjectproxy.h" "../src/glib-2-cc73e48deb.clean/gio/gdbusobjectmanager.h" "../src/glib-2-cc73e48deb.clean/gio/gdbusobjectmanagerclient.h" "../src/glib-2-cc73e48deb.clean/gio/gdbusobjectmanagerserver.h" "../src/glib-2-cc73e48deb.clean/gio/gtestdbus.h" "gio/gnetworking.h"
ninja: build stopped: subcommand failed.
```
</details>

**Additional context**

`gversionmacros.h` presents in the `buildtrees/glib/x86-windows-dbg/glib`, but not in the `buildtrees/glib/x86-windows-rel/glib`.

```
[17/479] cl.exe ... ../src/glib-2-cc73e48deb.clean/gio/win32/gwinhttpfile.c
FAILED: gio/win32/libgiowin32.a.p/gwinhttpfile.c.obj 
fatal error C1083: Cannot open include file: 'glib/gversionmacros.h': No such file or directory
```

This one's a race condition in the upstream makefiles, between the gio/win32/ and glib/ subdirectories. My debugging is in https://github.com/wesnoth/wesnoth/pull/7352#issuecomment-1415871708 (that PR isn't a fix, it was just to store the logs from the CI).

I think this one hasn't been logged or fixed upstream yet; but as I don't do Windows builds locally, I'm not the person to take it upstream.

--- a/ports/glib/fix-build-race-on-gio.patch
+++ b/ports/glib/fix-build-race-on-gio.patch
@@ -1,0 +1,24 @@
+diff --git a/gio/kqueue/meson.build b/gio/kqueue/meson.build
+index 7447e56..f3e573f 100644
+--- a/gio/kqueue/meson.build
++++ b/gio/kqueue/meson.build
+@@ -10,6 +10,7 @@ kqueue_lib = static_library('kqueue',
+   include_directories : [configinc, glibinc],
+   dependencies : [
+     gioenumtypes_dep,
++    libglib_dep,
+     gmodule_inc_dep,
+   ],
+   gnu_symbol_visibility : 'hidden',
+diff --git a/gio/win32/meson.build b/gio/win32/meson.build
+index 08be6b0..6699f10 100644
+--- a/gio/win32/meson.build
++++ b/gio/win32/meson.build
+@@ -13,6 +13,7 @@ giowin32_lib = static_library('giowin32',
+   dependencies : [
+     libintl,
+     gioenumtypes_dep,
++    libglib_dep,
+     gmodule_inc_dep,
+   ],
+   gnu_symbol_visibility : 'hidden',

--- a/ports/glib/portfile.cmake
+++ b/ports/glib/portfile.cmake
@@ -10,6 +10,7 @@ vcpkg_extract_source_archive(SOURCE_PATH
     PATCHES
         use-libiconv-on-windows.patch
         libintl.patch
+        fix-build-race-on-gio.patch # https://gitlab.gnome.org/GNOME/glib/-/merge_requests/3512
 )
 
 vcpkg_list(SET OPTIONS)

--- a/ports/glib/vcpkg.json
+++ b/ports/glib/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "glib",
   "version": "2.76.3",
-  "port-version": 1,
+  "port-version": 2,
   "description": "Portable, general-purpose utility library.",
   "homepage": "https://developer.gnome.org/glib/",
   "license": "LGPL-2.1-or-later",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -2890,7 +2890,7 @@
     },
     "glib": {
       "baseline": "2.76.3",
-      "port-version": 1
+      "port-version": 2
     },
     "glibmm": {
       "baseline": "2.76.0",

--- a/versions/g-/glib.json
+++ b/versions/g-/glib.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "c4aa04cd3294219ddd4b43c2f02b95c100683156",
+      "version": "2.76.3",
+      "port-version": 2
+    },
+    {
       "git-tree": "1d1c3788ae7982f573db3e59116dd1ca885b88a4",
       "version": "2.76.3",
       "port-version": 1

